### PR TITLE
read.js - fix handling of query limits

### PR DESCRIPTION
**Pull request by bkutil**
There was a bug in read that caused only 1k of points to be fetched by
default.

Update the limit-handling code so that:

1. User choices override optimization settings

2. When base class calls `read()` to fetch more points, we keep track
   of points fetched, and compute the limit as a minimum from the
   `batchSize` parameter passed to us and the remaining points to be
   read in total.

### CODE REVIEW

**Comment by bkutil** [comment]
cc: @demmer @dmehra

**Comment by bkutil** [comment]
We talked with @demmer about this via hangouts, here are the notes:

- optimization defines the limit
- there should be an internal max batch limit per adapter (say, 10k points - make it configurable in adapter config), and don't rely on the scheduler not being evil
- otherwise obey whatever the says to fetch within out internal max batch limit

File: lib/read.js [hunk 1]
```diff
@@ -136,13 +139,17 @@ class ReadInflux extends AdapterRead {
...
+    read(from, to, batchSize, state) {
         var self = this;
 
         var parsedUrl = _.clone(this.url);
         var reqUrl = null;
 
-        var queryOptions = _.extend({ to, from }, this.queryOptions);
+        // If scheduler is throttling our reads via batchSize param, take it,
+        // even if there are more points left to read.
+        var limit = Math.min(batchSize, this.queryOptions.limit - this.pointsRead);
```
Comment (VladVega) [inline lib/read.js]: I think most other adapters use the `fetchSize` option. Consider renaming to the `limit` option here to `fetchSize`.

File: lib/read.js [hunk 2]
```diff
@@ -136,13 +139,17 @@ class ReadInflux extends AdapterRead {
...
+
+        var queryOptions = _.defaults({ to, from, limit }, this.queryOptions);
```
Comment (VladVega) [inline lib/read.js]: Consider not defaulting `to` and `from

**Comment by VladVega** [comment]
Besides the 2 comments above this looks fine, +1.

**Comment by bkutil** [comment]
I renamed the attribute to `fetchSize` and made sure that options won't contain `from` or `to`. Thanks!

Will merge on green.

**Pull request merged**